src/convertToObject.js

```diff
@@ -6,7 +6,17 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .filter((line) => line.trim())
+    .map((line) => line.split(/:(.+)/).map((part) => part.trim()))
```

Comment:
The regular expression `/(.+)/` in `line.split(/:(.+)/)` is intended to split the string into a key-value pair. However, it may not handle cases where there are multiple colons correctly. Consider using a more precise regular expression or method to ensure only the first colon is used for splitting.